src/toggle-button/style.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { SYSTEM } from '../internal/environment';
+import customCssProps from '../internal/generated/custom-css-properties';
+import { ToggleButtonProps } from './interfaces';
+
+export function getToggleButtonStyles(style: ToggleButtonProps['style']) {
+  if (SYSTEM !== 'core') {
+    return {};
+  }
+
+  const properties = {
+    borderRadius: style?.root?.borderRadius,
+    borderWidth: style?.root?.borderWidth,
+    paddingBlock: style?.root?.paddingBlock,
+    paddingInline: style?.root?.paddingInline,
+    [customCssProps.styleBackgroundActive]: style?.root?.background?.active,
+    [customCssProps.styleBackgroundDefault]: style?.root?.background?.default,
+    [customCssProps.styleBackgroundDisabled]: style?.root?.background?.disabled,
+    [customCssProps.styleBackgroundHover]: style?.root?.background?.hover,
+    [customCssProps.styleBackgroundPressed]: style?.root?.background?.pressed,
+    [customCssProps.styleBorderColorActive]: style?.root?.borderColor?.active,
+    [customCssProps.styleBorderColorDefault]: style?.root?.borderColor?.default,
+    [customCssProps.styleBorderColorDisabled]: style?.root?.borderColor?.disabled,
+    [customCssProps.styleBorderColorHover]: style?.root?.borderColor?.hover,
+    [customCssProps.styleBorderColorPressed]: style?.root?.borderColor?.pressed,
+    [customCssProps.styleBoxShadowActive]: style?.root?.boxShadow?.active,
+    [customCssProps.styleBoxShadowDefault]: style?.root?.boxShadow?.default,
+    [customCssProps.styleBoxShadowDisabled]: style?.root?.boxShadow?.disabled,
+    [customCssProps.styleBoxShadowHover]: style?.root?.boxShadow?.hover,
+    [customCssProps.styleBoxShadowPressed]: style?.root?.boxShadow?.pressed,
+    [customCssProps.styleColorActive]: style?.root?.color?.active,
+    [customCssProps.styleColorDefault]: style?.root?.color?.default,
+    [customCssProps.styleColorDisabled]: style?.root?.color?.disabled,
+    [customCssProps.styleColorHover]: style?.root?.color?.hover,
+    [customCssProps.styleColorPressed]: style?.root?.color?.pressed,
+    [customCssProps.styleFocusRingBorderColor]: style?.root?.focusRing?.borderColor,
+    [customCssProps.styleFocusRingBorderRadius]: style?.root?.focusRing?.borderRadius,
+    [customCssProps.styleFocusRingBorderWidth]: style?.root?.focusRing?.borderWidth,
+  };
+
+  return Object.fromEntries(Object.entries(properties).filter(([, value]) => value));
```

Comment:
why do we need this?

Reply:
We filter out undefined properties with that, to prevent any unwanted side effects. I am changing it to be more explicit:

```
return Object.fromEntries(Object.entries(properties).filter(([, value]) => value !== undefined));
```

Reply:
What side effects, for example? I noticed that we don't do that e.g. in the button's style utils. If there is a value - let's do it uniformly, and for that a shared utility would be nice to have. If there is no real value - let's not do that.

Reply:
Discussion moved here: https://github.com/cloudscape-design/components/pull/4025#discussion_r2523574346